src/wallets/IdentityWallet.ts

```diff
@@ -150,128 +128,86 @@ export class IdentityWallet implements TLWallet {
   }
 
   /**
-   * Decrypts given serialized wallet and loads wallet.
-   * @param serializedWallet serialized wallet from `createAccount`.
-   * @param password Password to decrypt wallet.
-   * @param progressCallback Callback function for decryption progress.
+   * Loads given identity wallet.
+   * @param identityWallet `TLWallet` of type `WALLET_TYPE_IDENTITY`.
    */
   public async loadAccount(
-    serializedWallet: string,
-    password: string,
-    progressCallback?: any
-  ): Promise<UserObject> {
-    const deserializedWallet: IdentityWalletSchema = JSON.parse(
-      serializedWallet
+    identityWallet: IdentityWalletSchema
+  ): Promise<void> {
+    verifyWalletTypeAndVersion(
+      identityWallet,
+      WALLET_TYPE_IDENTITY,
+      EXPECTED_VERSIONS
     )
 
-    this.verifyDeserializedWalletHandled(deserializedWallet)
-
-    const encryptedKeystore = deserializedWallet.ethersKeystore
-    const identityAddress = deserializedWallet.identityAddress
-
-    this.wallet = await ethers.Wallet.fromEncryptedJson(
-      encryptedKeystore,
-      password,
-      typeof progressCallback === 'function' && progressCallback
+    const walletFromEthers = utils.getWalletFromEthers(identityWallet)
+    this.walletFromEthers = walletFromEthers
+    this.identityAddress = calculateIdentityAddress(
```

Comment:
I would not recalculate the address here, because that might be slow. 
Lets just store it.